### ISSUE TITLE
Add on-demand random salt

### ISSUE THREAD

**Issue by makersphereLabs** [makersphereHQ/emoji-passcodes#13]
In order to make the application even more secure, we could add something like a JavaScript based **build script**, which creates a random salt **for every user**. The fresh generated salt then gets baked into the Bookmarklet.

If we want to make this work, we need to develop a new landing page.
Jekyll would be a good option. 

💡 https://github.com/makersphereHQ/emoji-passcodes/pull/8#issuecomment-249229476

**Comment by atoponce** [comment]
One drawback to this approach, is people with multiple browsers, on one computer, or across many computers, such as at work and at home. If JavaScript is creating a random salt for each build, the every bookmarklet will be different, which means Johnny's bookmarklet in Firefox at home will be different than his bookmarklet in Firefox at work.

Instead, an input text field asking for an identifier, unique to only them, such as an email address, would be best. Checkout miniLock as an example of how this is done.

**Comment by makersphereLabs** [comment]
@atoponce You're right, that is an issue. Some users sync their bookmarks across devices (with Firefox Sync, iCloud, etc.), but this can also be a huge attack surface. We can make the Bookmarklet portable (e.g. export feature), but it's very unlikely that "normal" users would use this feature. 

The way miniLock does it looks promising, but if someone knows your email address... 🤔
Hmm, @DaKnOb What do you think about this approach?

**Comment by atoponce** [comment]
It's not so much that the salt is secret, as much as it is unique per user. The only goal here is to prevent precomputed tables, which unique salts per user address. It would be best if they were truly randomized, though. But, uniqueness is the ultimate key.

**Comment by makersphereLabs** [comment]
Yep, this system prevents precomputed tables but targeted attacks are still possible (let's ignore feasibility for a moment).
It depends on the threat model and _what kind of target you are_.
The good thing is, that you can enter **any email address** you like. Even not your real one.

**Comment by makersphereLabs** [comment]
@atoponce Alright, we've made some **huge** changes. 😉
Can you pull & test #16 in your favorite browsers?

**Comment by atoponce** [comment]
Yeah. That looks good. I submitted a pull request to use the HTML5 email form type, instead of text. This will provide automatic email validation, and on mobile devices, keyboards will present a ".com" key for auto-completing the email.

**Comment by makersphereLabs** [comment]
@atoponce Nice! Totally missed that. 😅
Mobile devices are currently _not fully supported_. (e.g. responsive design) #15
We need to modify the design and conduct "field tests" with multiple devices (iOS/Android).
What about a mobile-only version with offline support, etc.?

**Comment by atoponce** [comment]
That would be nice. I was also thinking, it may be best to just bundle the scrypt library with the code, rather than download it from CDNJS. It is 1.2 MB in size, so it bloats the bookmarklet/app a bit, but then can be run offline.
